torchrl/collectors/distributed/distributed_collector.py

```diff
@@ -0,0 +1,232 @@
+import torch
+import logging
+from abc import ABC
+from typing import Iterator
+from torch.utils.data import IterableDataset
+from tensordict.tensordict import TensorDictBase
+from ray._private.services import get_node_ip_address
+
+
+logger = logging.getLogger(__name__)
+
+try:
+    import ray
+    _has_ray = True
+except ImportError as err:
+    _has_ray = False
+
+default_remote_config = {
+    "num_cpus": 1,
+    "num_gpus": 0.2,
+    "memory": 5 * 1024 ** 3,
+    "object_store_memory": 2 * 1024 ** 3
+}
+
+
+def print_remote_collector_info(self):
+    """Print some information about the remote collector."""
+    s = f"Created remote collector in machine {get_node_ip_address()} using gpus {ray.get_gpu_ids()}"
+    # logger.warning(s)
+    print(s)
+
+
+@classmethod
+def as_remote(cls,
+              num_cpus=None,
+              num_gpus=None,
+              memory=None,
+              object_store_memory=None,
+              resources=None):
+    """
+    Creates an instance of a remote ray class.
+
+    Parameters
+    ----------
+    cls: Python Class
+        Class to be remotely instantiated.
+    num_cpus : int
+        The quantity of CPU cores to reserve for this class.
+    num_gpus  : float
+        The quantity of GPUs to reserve for this class.
+    memory : int
+        The heap memory quota for this class (in bytes).
+    object_store_memory : int
+        The object store memory quota for this class (in bytes).
+    resources: Dict[str, float]
+        The default resources required by the class creation task.
+
+    Returns
+    -------
+    w : func
+        A function to create ray remote class instances.
+    """
+    c = ray.remote(
+        num_cpus=num_cpus,
+        num_gpus=num_gpus,
+        memory=memory,
+        object_store_memory=object_store_memory,
+        resources=resources)(cls)
+    c.is_remote = True
+    return c
+
+
+class DistributedCollector(IterableDataset, ABC):
+    """
+    Class to better handle the operations of ensembles of Collectors.
+
+    Contains common functionality across all collectors sets.
```

Comment:
This is not very explicit.
We should say 
"Like other collectors, this collector is an iterable that yields TensorDict instances with the appropriate number of frames."

We should say what is and what is not supported.

We should say if it's sync, async or depends on the input to the constructor.